vault/src/api.rs

```diff
@@ -55,10 +55,28 @@ fn handle_resp<T: Encode>(resp: Result<T, Error>) -> Result<Value, JsonRpcError>
     }
 }
 
-fn _system_health(provider: &Arc<PolkaBtcProvider>) -> Result<(), Error> {
-    match block_on(timeout(HEALTH_DURATION, provider.get_latest_block_hash())) {
-        Err(err) => Err(Error::RuntimeError(RuntimeError::from(err))),
-        _ => Ok(()),
+async fn _system_health(provider: &Arc<PolkaBtcProvider>) -> Result<(), Error> {
+    let result = match timeout(HEALTH_DURATION, provider.get_latest_block()).await {
+        Ok(res) => res,
+        Err(err) => return Err(Error::RuntimeError(RuntimeError::from(err))),
+    };
+
+    let signed_block = result?.ok_or(Error::NoIncomingBlocks)?;
+    let current_time: u128 = signed_block.block.header.number.into();
+
+    // TODO: parameterize based on expiry
+    let has_uncompleted =
```

Comment:
we should probably check how long we have been running, since we want to give the vault time to execute the redeems (otherwise it would get stuck forever). Also we should check if the request was expired but not cancelled, since the vault shouldn't execute the redeem in that case

Reply:
added expiry check, we can wait before calling endpoint in-cluster